src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -267,8 +267,7 @@ trait Infer extends Checkable {
         case NoSymbol if sym.isJavaDefined && context.unit.isJava => sym  // don't try to second guess Java; see #4402
         case sym1                                                 => sym1
       }
-      // XXX So... what's this for exactly?
-      if (context.unit.exists)
+      if (settings.YtrackDependencies.value)
```

Comment:
`if (settings.YtrackDependencies)` but I like how it answers the comment question